frontend/catalyst/utils/toml.py

```diff
@@ -82,13 +125,18 @@ def check_quantum_control_flag(config: TOMLDocument) -> bool:
     raise CompileError("quantum_control flag is not supported in TOMLs schema >= 2")
 
 
-def get_gates(config: TOMLDocument, path: List[str], shots_present: bool) -> Dict[str, dict]:
+def get_gates(
```

Comment:
This function is not very clear, can you document the arguments? To me it seems that conditions are only relevant for observables

Reply:
Ok, will do. The idea is to load a toml-schema section while taking condition properties into account, by comparing them with the `ProgramFeatures` we already know from user.

Reply:
Done